src/main/java/com/crystal/ManFight.java

```diff
@@ -71,34 +71,34 @@ public void startFight() {
         scanner.close();
     }
 
-    public boolean doMove(String move, Man manAttacking, Man manBeingAttacked) {
+    public boolean doMove(String move, Character manAttacking, Character characterBeingAttacked) {
         if (manAttacking.isScared()) {
             System.out.println(manAttacking.getName() + " is still scared! can't move!");
             manAttacking.setScared(false);
             return true;
         }
-        if (manAttacking.getManClass() == ManClass.MAGICIAN) {
-            boolean hasMagicWand = doMoveAsMagician(move, manAttacking);
+        if (manAttacking instanceof Magician) {
+            boolean hasMagicWand = doMoveAsMagician(move, (Magician) manAttacking);
             if (!hasMagicWand) {
                 return false;
             }
         }
         switch (move) {
             case "kick":
-                return manAttacking.kick(manBeingAttacked);
+                return manAttacking.kick(characterBeingAttacked);
             case "punch":
-                return manAttacking.punch(manBeingAttacked);
+                return manAttacking.punch(characterBeingAttacked);
             case "headbutt":
-                return manAttacking.headbutt(manBeingAttacked);
+                return manAttacking.headbutt(characterBeingAttacked);
             case "roar":
-                return manAttacking.roar(manBeingAttacked);
+                return manAttacking.roar(characterBeingAttacked);
             default:
                 System.out.println(move + " is not a valid move you skip your turn doofus!");
                 return true;
         }
     }
 
-    public boolean doMoveAsMagician(String move, Man manAttacking) {
+    public boolean doMoveAsMagician(String move, Magician manAttacking) {

```

Comment:
I'm not a huge fan of this, I would rather keep this logic inside the individual character classes, maybe have a an attack method?

Reply:
done